src/components/Facet.vue

```diff
@@ -57,14 +57,31 @@ export default {
 
       this.$router.push({ query: newQuery });
     },
+    updateFacetsFromURL: function (query) {
+      this.checkedFacets = [];
+      for (const param in query) {
+        if (this.facetHeader == param && typeof query[param] == "string") {
+          this.checkedFacets.push(query[param]);
+        } else if (this.facetHeader == param && Array.isArray(query[param])) {
+          query[param].forEach((facet) => {
+            this.checkedFacets.push(facet);
+          });
+        }
+      }
+    },
   },
   mounted() {
-    this.$watch(
-      () => this.checkedFacets,
-      () => {
-        this.applyFacets();
-      }
-    );
+    if (this.$route.query) {
+      this.updateFacetsFromURL(this.$route.query);
+    }
+  },
+  watch: {
+    $route(to) {
```

Comment:
We need the route watcher because the component doesn't re-mount on browser history changes (at least, not as far as I can tell). However, it's possible that we can remove the `mounted()` hook, or change it to a different lifecycle hook. I'll take a look and follow up with you on Slack.

Reply:
Ah, browser history was the bit I didn't consider. Weird but also 👍🏻